test/inductor/test_triton_kernels.py

```diff
@@ -3550,6 +3550,71 @@ def f(x, increment):
 
         self.assertEqual(y + increment, x)
 
+    @requires_gpu
+    @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
+    @common_utils.parametrize("autotune_at_compile_time", [True, False])
+    @common_utils.parametrize("with_perf_model", [False, True])
+    def test_triton_kernel_prune_configs_by(
+        self, backend, autotune_at_compile_time, with_perf_model
+    ):
+        N = 1024
+        src = torch.randn(N, device=GPU_TYPE)
+        dst = torch.empty(N, device=GPU_TYPE)
+        records = {}
+
+        def early_config_prune(configs, named_args, **kwargs):
+            records["run_early_config_prune"] = True
+            if "N" in kwargs and kwargs["N"] == 1024:
+                records["capture_kwargs"] = True
+            elif "N" in kwargs:
+                breakpoint()
+            if "dst" in named_args and "src" in named_args and len(named_args) == 2:
+                records["capture_named_args"] = True
+            return [configs[0]]
+
+        def perf_model(*args, **kwargs):
+            records["run_perf_model"] = True
+            return kwargs["BLOCK_SIZE"]
+
+        configs = [
+            triton.Config(kwargs={"BLOCK_SIZE": 32}),
+            triton.Config(kwargs={"BLOCK_SIZE": 128}),
+        ]
+
+        if with_perf_model:
+            prune_configs_by = {"perf_model": perf_model, "top_k": 1}
+        else:
+            prune_configs_by = {"early_config_prune": early_config_prune}
+
+        @triton.autotune(configs=configs, key=["N"], prune_configs_by=prune_configs_by)
```

Comment:
`key` is ignored by the current implementation in PT2, so better just leave out to avoid confusion.